src/utils/daos_hdlr.c

```diff
@@ -2646,45 +2648,421 @@ set_dm_args_default(struct dm_args *dm)
+	props->dpp_entries[8].dpe_type = DAOS_PROP_CO_SNAPSHOT_MAX;
+	props->dpp_entries[9].dpe_type = DAOS_PROP_CO_COMPRESS;
+	props->dpp_entries[10].dpe_type = DAOS_PROP_CO_ENCRYPT;
+	props->dpp_entries[11].dpe_type = DAOS_PROP_CO_OWNER;
+	props->dpp_entries[12].dpe_type = DAOS_PROP_CO_OWNER_GROUP;
+	props->dpp_entries[13].dpe_type = DAOS_PROP_CO_DEDUP;
+	props->dpp_entries[14].dpe_type = DAOS_PROP_CO_DEDUP_THRESHOLD;
+
+	/* Conditionally get the OID. Should always be true for serialization. */
+	if (get_oid) {
+		props->dpp_entries[15].dpe_type = DAOS_PROP_CO_ALLOCED_OID;
+	}
+
+	/* Get all props except ACL first. */
+	rc = daos_cont_query(coh, NULL, props, NULL);
+	if (rc != 0) {
+		fprintf(stderr, "Failed to query container: "DF_RC, DP_RC(rc));
+		rc = 1;
+		D_GOTO(out, rc);
+	}
+
+	/* Fetch the ACL separately in case user doesn't have access */
+	rc = daos_cont_get_acl(coh, &prop_acl, NULL);
+	if (rc == 0) {
+		/* ACL will be appended to the end */
+		props_merged = daos_prop_merge(props, prop_acl);
+		if (props_merged == NULL) {
+			fprintf(stderr, "Failed to set container ACL: "DF_RC,
+				DP_RC(rc));
+			rc = 1;
+			D_GOTO(out, rc);
+		}
+		daos_prop_free(props);
+		props = props_merged;
+	} else if (rc && rc != -DER_NO_PERM) {
+		fprintf(stderr, "Failed to query container ACL: "DF_RC,
+			DP_RC(rc));
+		rc = 1;
+		D_GOTO(out, rc);
+	}
+
+	rc = 0;
+	*_props = props;
+	
+out:
+	daos_prop_free(prop_acl);
+	if (rc != 0) {
+		daos_prop_free(props);
+	}
+
+	return rc;
+}
+
+static int
+dm_serialize_metadata(struct dm_args *ca, daos_prop_t *props, char *preserve)
+{
+	int	rc = 0;
+	int	num_attrs = 0;
+	char	**names = NULL;
+	void	**buffers = NULL;
+	size_t	*sizes = NULL;
+	void	*handle;
+	
+	int (*serialize_daos_metadata)(char *, daos_prop_t *props,
+				       daos_handle_t, int, char **,
+				       char **, size_t *);
+	/* Get all user attributes if any exist */
+	rc = dm_cont_get_usr_attrs(ca->src_coh, &num_attrs, &names,
+				   &buffers, &sizes);
+	if (rc != 0) {
+		rc = 1;
+		D_GOTO(out, rc);
+	}
+
+	handle = dlopen(LIBSERIALIZE, RTLD_NOW);
+	if (handle == NULL) {
+		rc = EINVAL;
+		D_GOTO(out, rc);
+	}
+	serialize_daos_metadata = dlsym(handle, "serialize_daos_metadata");
+	if (serialize_daos_metadata == NULL)  {
+		rc = EINVAL;
+		D_GOTO(out, rc);
+	}
+	(*serialize_daos_metadata)(preserve, props, ca->src_coh, num_attrs,
+				   names, (char**)buffers, sizes);
+out:
+	if (num_attrs > 0) {
+		dm_cont_free_usr_attrs(num_attrs, &names, &buffers, &sizes);
+	}
+	return rc;
+}
+
+static int
+dm_deserialize_cont_prop_metadata(struct dm_args *ca, char *preserve,
+				  daos_prop_t *props)
+{
+	int	rc = 0;
+	void	*handle;
+
+	int (*deserialize_daos_cont_prop_metadata)(char *,
+				         	   daos_prop_t *props,
+					 	   uint64_t *);
+	handle = dlopen(LIBSERIALIZE, RTLD_NOW);
+	if (handle == NULL) {
+		rc = EINVAL;
+		D_GOTO(out, rc);
+	}
+	deserialize_daos_cont_prop_metadata = dlsym(handle, "deserialize"
+					  	    "_daos_cont_prop_metadata");
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs
(style)  break quoted strings at a space character